ring/bench/ring_memberlist_test.go

```diff
@@ -0,0 +1,100 @@
+package bench
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/go-kit/log"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/stretchr/testify/require"
+
+	"github.com/grafana/dskit/flagext"
+	"github.com/grafana/dskit/kv/codec"
+	"github.com/grafana/dskit/kv/memberlist"
+	"github.com/grafana/dskit/ring"
+	"github.com/grafana/dskit/services"
+)
+
+type dnsProviderMock struct {
+	resolved []string
+}
+
+func (p *dnsProviderMock) Resolve(ctx context.Context, addrs []string) error {
+	p.resolved = addrs
+	return nil
+}
+
+func (p dnsProviderMock) Addresses() []string {
+	return p.resolved
+}
+
+func encodeMessage(b *testing.B, key string, d *ring.Desc) []byte {
+	c := ring.GetCodec()
+	val, err := c.Encode(d)
+	require.NoError(b, err)
+
+	kvPair := memberlist.KeyValuePair{
+		Key:   key,
+		Value: val,
+		Codec: c.CodecID(),
+	}
+
+	ser, err := kvPair.Marshal()
+	require.NoError(b, err)
+	return ser
+}
+
+// Benchmark the memberlist receive path when it us being used as the ring backing store.
+func BenchmarkMemberlistReceiveWithRingDesc(b *testing.B) {
+	c := ring.GetCodec()
+
+	var cfg memberlist.KVConfig
+	flagext.DefaultValues(&cfg)
+	cfg.TCPTransport = memberlist.TCPTransportConfig{
+		BindAddrs: []string{"localhost"},
+	}
+	cfg.Codecs = []codec.Codec{c}
+
+	mkv := memberlist.NewKV(cfg, log.NewNopLogger(), &dnsProviderMock{}, prometheus.NewPedanticRegistry())
+	require.NoError(b, services.StartAndAwaitRunning(context.Background(), mkv))
+	defer services.StopAndAwaitTerminated(context.Background(), mkv) //nolint:errcheck
+
+	_, err := memberlist.NewClient(mkv, c)
+	require.NoError(b, err)
```

Comment:
i think it's not obvious (to me) what this is for, is it just to verify that `mkv` can get the codec? I think it might make sense to add a comment explaining why this is necessary.

Reply:
It's actually not needed, good spot. Copy/paste from the test I based it off.